homeassistant/components/cover/brunt.py

```diff
@@ -0,0 +1,201 @@
+from homeassistant.components.cover import (
+    ATTR_CURRENT_POSITION, ATTR_POSITION,
+    CoverDevice, PLATFORM_SCHEMA,
+    STATE_CLOSED, STATE_CLOSING,
+    STATE_OPEN, STATE_OPENING,
+    STATE_UNKNOWN, SUPPORT_CLOSE,
+    SUPPORT_OPEN, SUPPORT_SET_POSITION
+)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['brunt==0.1.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTRIBUTION = 'Based on an unofficial Brunt SDK.'
+
+COVER_FEATURES = SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_SET_POSITION
+
+STATE_PARTIALLY_OPEN = 'partially open'
+ATTR_REQUEST_POSITION = 'request_position'
+DEFAULT_NAME = 'brunt blind engine'
+NOTIFICATION_ID = 'brunt_notification'
+NOTIFICATION_TITLE = 'Brunt Cover Setup'
+
+CLOSED_POSITION = 0
+OPEN_POSITION = 100
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string
+})
+
+POSITION_MAP = {
+    0: STATE_CLOSED,
+    100: STATE_OPEN,
+}
+MOVE_STATES_MAP = {
+    1: STATE_OPENING,
+    2: STATE_CLOSING,
+}
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the brunt platform."""
+    # pylint: disable=no-name-in-module
+    from brunt import BruntAPI
+    username = config[CONF_USERNAME]
+    password = config[CONF_PASSWORD]
+
+    bapi = BruntAPI(username=username, password=password)
+    try:
+        things = bapi.getThings()['things']
+        if not things:
+            _LOGGER.error("No things present in account.")
+        else:
+            add_devices(BruntDevice(
+                hass, bapi, thing['NAME'],
+                thing['thingUri']) for thing in things)
+    except (TypeError, KeyError, NameError, ValueError) as ex:
+        _LOGGER.error("%s", ex)
+        hass.components.persistent_notification.create(
+            'Error: {}<br />'
+            'You will need to restart hass after fixing.'
+            ''.format(ex),
+            title=NOTIFICATION_TITLE,
+            notification_id=NOTIFICATION_ID)
+
+
+class BruntDevice(CoverDevice):
+    """
+    Representation of a Brunt cover device.
+
+    Contains the common logic for all Brunt devices.
+    """
+
+    def __init__(self, hass, bapi, name, thing_uri):
+        """Init the Brunt device."""
+        self._bapi = bapi
+        self._name = name
+        self._thing_uri = thing_uri
+
+        self._state = STATE_UNKNOWN
+        self._available = None
+        self.update()
+
+    @property
+    def name(self):
+        """Return the name of the device as reported by tellcore."""
+        return self._name
+
+    @property
+    def available(self):
+        """Could the device be accessed during the last update call."""
+        return self._available
+
+    @property
+    def current_cover_position(self):
+        """
+        Return current position of cover.
+
+        None is unknown, 0 is closed, 100 is fully open.
+        """
+        return int(self._state['currentPosition'])
```

Comment:
This will error as long as state is not populated. Use eg `dict.get` with a safe default instead and make the integer copy conditional.